spark/src/test/java/org/apache/iceberg/spark/actions/TestNewRewriteDataFilesAction.java

```diff
@@ -590,6 +595,156 @@ public void testInvalidOptions() {
             .execute());
   }
 
+  @Test
+  public void testSortMultipleGroups() {
+    Table table = createTable(20);
+    shouldHaveFiles(table, 20);
+    table.replaceSortOrder().asc("c2").commit();
+    int fileSize = averageFileSize(table);
+
+    List<Object[]> originalData = currentData();
+
+    // Perform a rewrite but only allow 2 files to be compacted at a time
+    RewriteDataFiles.Result result =
+        basicRewrite(table)
+            .sort()
+            .option(SortStrategy.REWRITE_ALL, "true")
+            .option(RewriteDataFiles.MAX_FILE_GROUP_SIZE_BYTES, Integer.toString(fileSize * 2 + 100))
+            .option(BinPackStrategy.MIN_INPUT_FILES, "1")
+            .execute();
+
+    Assert.assertEquals("Should have 10 fileGroups", result.rewriteResults().size(), 10);
+
+    table.refresh();
+
+    List<Object[]> postRewriteData = currentData();
+    assertEquals("We shouldn't have changed the data", originalData, postRewriteData);
+
+    shouldHaveSnapshots(table, 2);
+    shouldHaveACleanCache(table);
```

Comment:
Shouldn't this validate that the table only has 10 files now and that the files are sorted by c2?

Reply:
Looks like this is probably because you don't want to use `shouldHaveLastCommitSorted(table, "c2");`?

Reply:
Ah yeah the result of this would not be completely sorted since it groups together 2 files at a time (pretty much at random) so the results shouldn't be globally sorted